src/Microsoft.Azure.WebJobs.Host.Storage/StorageServiceCollectionExtensions.cs

```diff
@@ -2,90 +2,59 @@
-            services.TryAddSingleton<DistributedLockManagerContainerProvider>();
-
-            services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<StorageAccountOptions>, StorageAccountOptionsSetup>());
             services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<JobHostInternalStorageOptions>, CoreWebJobsOptionsSetup<JobHostInternalStorageOptions>>());
 
             services.TryAddSingleton<IDelegatingHandlerProvider, DefaultDelegatingHandlerProvider>();
 
+            // May need to rename this to HostBlobServiceClientProvider
+            services.TryAddSingleton<BlobServiceClientProvider>();
+            services.AddAzureClientsCore();
+            services.TryAddSingleton<IAzureStorageProvider, AzureStorageProvider>();
+
             services.AddSingleton<IConcurrencyStatusRepository, BlobStorageConcurrencyStatusRepository>();
         }
 
         // This is only called if the host didn't already provide an implementation
+        // TODO: Should this attempt to create a client or just check for connection setting?
         private static IDistributedLockManager Create(IServiceProvider provider)
         {
-            // $$$ get rid of LegacyConfig
-            var opts = provider.GetRequiredService<IOptions<StorageAccountOptions>>();
-
+            var azureStorageProvider = provider.GetRequiredService<IAzureStorageProvider>();
             var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
-
-            var sas = provider.GetService<DistributedLockManagerContainerProvider>();
-
-            CloudBlobContainer container;
-
-            if (sas != null && sas.InternalContainer != null)
+            var logger = loggerFactory.CreateLogger<IDistributedLockManager>();
+            try
             {
-                container = sas.InternalContainer;
+                var container = azureStorageProvider.GetWebJobsBlobContainerClient();
+                logger.LogDebug("Using BlobLeaseDistributedLockManager in Functions Host.");
+                return new BlobLeaseDistributedLockManager(loggerFactory, azureStorageProvider);
             }
-            else
+            catch (InvalidOperationException)
             {
-                var config = opts.Value;
-                CloudStorageAccount account = config.GetStorageAccount();
-                if (account == null)
-                {
-                    return new InMemoryDistributedLockManager();
-                }
-
-                var blobClient = new CloudBlobClient(account.BlobStorageUri, account.Credentials, config.DelegatingHandlerProvider?.Create());
-                container = blobClient.GetContainerReference(HostContainerNames.Hosts);
+                // If there is an error getting the container client,
+                // register an InMemoryDistributedLockManager.
+                logger.LogDebug("Using InMemoryDistributedLockManager in Functions Host.");
```

Comment:
Shouldn't be any "functions host" references in here. The WebJobs SDK is used outside of Functions.